src/autoscaler/integration/integration_broker_api_test.go

```diff
@@ -112,102 +113,109 @@ var _ = Describe("Integration_Broker_Api", func() {
+			AfterEach(func() {
+				//clear the binding
+				resp, err := unbindService(bindingId, appId, serviceInstanceId)
 				Expect(err).NotTo(HaveOccurred())
-				req.Header.Set("Content-Type", "application/json")
-				req.Header.Set("Authorization", "Basic "+brokerAuth)
-				resp, err := httpClient.Do(req)
+				Expect(resp.StatusCode).To(Equal(200))
+				Eventually(func() int {
+					return getNumberOfBinding(bindingId, appId, serviceInstanceId)
+				}).Should(Equal(0))
+
+			})
+			It("should return 201, save scaling rules to db and call scheduler", func() {
+				resp, err := bindService(bindingId, appId, serviceInstanceId, schedulePolicyJson)
 				Expect(err).NotTo(HaveOccurred())
 				Expect(resp.StatusCode).To(Equal(201))
-				Eventually(getNumberOfBinding).Should(Equal(1))
-				Eventually(getNumberOfPolicyJson).Should(Equal(1))
+				Eventually(func() int {
+					return getNumberOfBinding(bindingId, appId, serviceInstanceId)
+				}).Should(Equal(1))
+				Eventually(func() int {
+					return getNumberOfPolicyJson(appId)
+				}).Should(Equal(1))
 				Consistently(scheduler.ReceivedRequests).Should(HaveLen(1))
+
 			})
 		})
-		Context("Policy without schedules", func() {
+		Context("Invalid policy", func() {
 			BeforeEach(func() {
 				scheduler.RouteToHandler("PUT", regPath, ghttp.RespondWith(http.StatusOK, "successful"))
 
 			})
-			It("should return 201, save scaling rules to db and not call scheduler", func() {
-				policy := fmt.Sprintf(policyTemplate, testAppId, noSchedulePolicyJson)
-				req, err := http.NewRequest("PUT", fmt.Sprintf("http://127.0.0.1:%d/v2/service_instances/%s/service_bindings/%s", components.Ports["serviceBroker"], testServiceInstanceId, testBindingId), strings.NewReader(policy))
-				Expect(err).NotTo(HaveOccurred())
-				req.Header.Set("Content-Type", "application/json")
-				req.Header.Set("Authorization", "Basic "+brokerAuth)
-				resp, err := httpClient.Do(req)
+			It("should return 400, save no scaling rule to db and not call scheduler", func() {
+
+				resp, err := bindService(bindingId, appId, serviceInstanceId, invalidPolicyJson)
 				Expect(err).NotTo(HaveOccurred())
-				Expect(resp.StatusCode).To(Equal(201))
-				Eventually(getNumberOfBinding).Should(Equal(1))
-				Eventually(getNumberOfPolicyJson).Should(Equal(1))
+				Expect(resp.StatusCode).To(Equal(400))
+				Eventually(func() int {
+					return getNumberOfBinding(bindingId, appId, serviceInstanceId)
+				}).Should(Equal(0))
+				Eventually(func() int {
+					return getNumberOfPolicyJson(appId)
+				}).Should(Equal(0))
 				Consistently(scheduler.ReceivedRequests).Should(HaveLen(0))
 			})
 		})
-		Context("Invalid policy", func() {
+		Context("Api-server is down", func() {
 			BeforeEach(func() {
-				scheduler.RouteToHandler("PUT", regPath, ghttp.RespondWith(http.StatusOK, "successful"))
+				stopApiServer()
+				scheduler.RouteToHandler("PUT", regPath, ghttp.RespondWith(http.StatusInternalServerError, "error"))
```

Comment:
You need to make sure that the API server is actually down before leaving the BeforeEach.

I also noticed that the service broker => API has no timeouts on those requests.